package/src/handler.ts

```diff
@@ -329,6 +329,131 @@ export function referencesQueries({
     return [from('current repository'), from('other repositories')]
 }
 
+export function findDocstring({
+    definitionLine,
+    fileText,
+    commentStyle,
+    docstringIgnore,
+}: {
+    definitionLine: number
+    fileText: string
+    commentStyle?: CommentStyle
+    docstringIgnore?: RegExp
+}): string | undefined {
+    if (!commentStyle) {
+        return undefined
+    }
+
+    function findDocstringInLineComments({
+        lineRegex,
+        lines,
+    }: {
+        lineRegex: RegExp
+        lines: string[]
+    }): string[] | undefined {
+        const docLines = takeWhile(
+            dropWhile(lines, line =>
+                docstringIgnore ? docstringIgnore.test(line) : false
+            ),
+            line => new RegExp(/^\s*/.source + lineRegex.source).test(line)
+        ).map(line =>
+            line.replace(new RegExp(/^\s*/.source + lineRegex.source), '')
+        )
+        return docLines.length > 0 ? docLines : undefined
+    }
+
+    function findDocstringInBlockComment({
+        block: { startRegex, lineNoiseRegex, endRegex },
+        lines,
+    }: {
+        block: BlockCommentStyle
+        lines: string[]
+    }): string[] | undefined {
+        // ⚠️ Local mutation
```

Comment:
Why is the local mutation needed?

Reply:
IMO, it turned out to be easier to understand by writing imperative steps on a mutable array of lines.